src/utils.js

```diff
@@ -207,185 +286,279 @@ async function fetchJSON(modelPath, fileName, progressCallback = null, fatal = t
-        }
+  for (var i = 1; i < arr.length; ++i) {
+    if (arr[i] > max) {
+      maxIndex = i;
+      max = arr[i];
     }
+  }
 
-    return maxIndex;
+  return maxIndex;
 }
 
-
+/**
+ * Compute the softmax of an array of numbers.
+ *
+ * @param {number[]} arr - The array of numbers to compute the softmax of.
+ * @returns {number[]} The softmax array.
+ */
 function softmax(arr) {
-    // Compute the maximum value in the array
-    const max = Math.max(...arr);
+  // Compute the maximum value in the array
+  const max = Math.max(...arr);
 
-    // Compute the exponentials of the array values
-    const exps = arr.map(x => Math.exp(x - max));
+  // Compute the exponentials of the array values
+  const exps = arr.map((x) => Math.exp(x - max));
 
-    // Compute the sum of the exponentials
-    const sumExps = exps.reduce((acc, val) => acc + val, 0);
+  // Compute the sum of the exponentials
+  const sumExps = exps.reduce((acc, val) => acc + val, 0);
 
-    // Compute the softmax values
-    const softmaxArr = exps.map(x => x / sumExps);
+  // Compute the softmax values
+  const softmaxArr = exps.map((x) => x / sumExps);
 
-    return softmaxArr;
+  return softmaxArr;
 }
 
+/**
+ * Calculates the logarithm of the softmax function for the input array.
+ * @param {number[]} arr - The input array to calculate the log_softmax function for.
+ * @returns {number[]} - The resulting log_softmax array.
+ */
 function log_softmax(arr) {
-    // Compute the softmax values
-    const softmaxArr = softmax(arr);
+  // Compute the softmax values
+  const softmaxArr = softmax(arr);
 
-    // Apply log formula to each element
-    const logSoftmaxArr = softmaxArr.map(x => Math.log(x));
+  // Apply log formula to each element
+  const logSoftmaxArr = softmaxArr.map((x) => Math.log(x));
 
-    return logSoftmaxArr;
+  return logSoftmaxArr;
 }
 
+/**
+ * Escapes regular expression special characters from a string by replacing them with their escaped counterparts.
+ *
+ * @param {string} string - The string to escape.
+ * @returns {string} - The escaped string.
+ */
 function escapeRegExp(string) {
-    return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
+  return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"); // $& means the whole matched string
 }
 
+/**
+ * Get the top k items from an iterable, sorted by descending order
+ *
+ * @param {Array} items - The items to be sorted
+ * @param {number} [top_k=0] - The number of top items to return (default: 0 = return all)
+ * @returns {Array} - The top k items, sorted by descending order
+ */
 function getTopItems(items, top_k = 0) {
-    // if top == 0, return all
+  // if top == 0, return all
 
-    items = Array.from(items)
-        .map((x, i) => [i, x])            // Get indices ([index, score])
-        .sort((a, b) => b[1] - a[1])      // Sort by log probabilities
+  items = Array.from(items)
+    .map((x, i) => [i, x]) // Get indices ([index, score])
```

Comment:
I would rather have the comments aligned and it might be best to focus on JSDoc only in one PR, otherwise it becomes hard to judge